tensorflow_lite_support/cc/task/vision/landmark_detector.h

```diff
@@ -0,0 +1,127 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_LITE_SUPPORT_CC_TASK_VISION_LANDMARK_DETECTOR_H_
+#define TENSORFLOW_LITE_SUPPORT_CC_TASK_VISION_LANDMARK_DETECTOR_H_
+
+#include <memory>
+#include <vector>
+
+#include "absl/container/flat_hash_set.h"
+#include "absl/status/status.h"
+#include "tensorflow/lite/c/common.h"
+#include "tensorflow/lite/core/api/op_resolver.h"
+#include "tensorflow/lite/core/shims/cc/kernels/register.h"
+#include "tensorflow_lite_support/cc/port/integral_types.h"
+#include "tensorflow_lite_support/cc/port/statusor.h"
+#include "tensorflow_lite_support/cc/task/core/external_file_handler.h"
+#include "tensorflow_lite_support/cc/task/vision/core/base_vision_task_api.h"
+#include "tensorflow_lite_support/cc/task/vision/core/frame_buffer.h"
+#include "tensorflow_lite_support/cc/task/vision/proto/landmarks_proto_inc.h"
+#include "tensorflow_lite_support/cc/task/vision/proto/landmark_detector_options_proto_inc.h"
+
+namespace tflite {
+namespace task {
+namespace vision {
+
+// Performs landmark detection on images.
+//
+// The API expects a TFLite model with optional, but strongly recommended,
+// TFLite Model Metadata.
+//
+// Input tensor:
+//  (kTfLiteUInt8)
+//    - image input of size `[batch x height x width x channels]`.
+//    - batch inference is not supported (`batch` is required to be 1).
+//    - only RGB inputs are supported (`channels` is required to be 3).
+//    - `height` and  `width` are required to be (192,192).
+//    - Normalization is not required.
+// Output tensor:
+//  (kTfLiteFloat32) 
+//    - one output tensor with 4 dimensions `[1 x 1 x num_keypoints x 3]`,
+//      where, `num_keypoints` is required to be 1 and the inner array   
+//      representing y-x coordinates of keypoints withscore in the form 
+//      `[y, x, score]`.
+//
+// An example of such model can be found at:
+// https://tfhub.dev/google/lite-model/movenet/singlepose/lightning/tflite/int8/4

```

Comment:
int8 cannot be supported I think. Can you please give a link to the float model?

Reply:
Actually, I used the latest version, the uint8 model with metadata, and if I use the float model, I will get different results, at least in terms of prediction score. Though if you want I can use float model but I have to update the unit test files as well.